applications/Datagen_Project.md

```diff
@@ -245,3 +245,93 @@ The goal is to achive a fully functional mechanism for the random selection of t
+and we fell that a payment integration for the milestone will allow us to speed up the milestone development.
+
+I would also like to do some specification about the deliverable of the second milestone, just to underline the complexity of we
+weren’t completely aware when we submitted the original grant proposal.
+
+a) It revolves around the communication between a solo substrate chain and a parachain. Usually in the substrate framework you have a
+Relay Chain and a Parachain speaking with each other (due to the way in which the Polkadot and Kusama ecosystem are built), meaning
+that for the communication between a Parachain and a Solo Chain (which is not its own Relay Chain), the standard XCM method can’t be
+applied;
+
+b) For the same reasons you need custom RPC bridges;
+
+c) There is very fragmentary previous work; something can be found in Parity’s repo https://github.com/paritytech/solo-para-bridge-poc
+and some work has been done by Darwinia’s team, but all those repos are in beta, not ready to use (by us or by other teams);
+
+d) It would be the first time, best of my knowledge, that someone attempts to connect via RPC bridge a Parachain with a Solochain which
+is not the Parachain’s Relay;
+
+e) This implies more development;
+
+f) This can benefit a plurality of teams in the Polkadot ecosystem: for example it could enable direct communication between a
+Polkadot’s Parachain and Kusama (or viceversa) or any Polkadot’s/ Kusama’s Parachain and any Solochain; it can also enable Parachain’s
+to speak with Solochains which have a different blocktime;
+
+g) Solochain’s use can be beneficial for any use case in which is required a latency time lower than the ≈6 sec of blocktime required
+by Polkadot and can greatly expand Polkadot’s/Kusama’s use cases for low latency time applications; however the lack of frameworks for
+efficient communications between Solochains and Paracahains (which are the backbone of the Polkadot/Kusama ecosystem), is greatly
+reducing the use of Solochains as a complement for Parachains.
+
+h) Besides the importance of the development of the Milestone itself, we remember that Datagen (https://github.com/Datagen-Project/Whitepaper-2.0 ), as a service blockchain for verification of random computational problems, can greatly benefit the Polkadot
+network, by adding a tool to randomly verify on-chain off-chain computational workloads and to assign workloads in a decentralized way
+(as a long term solution on the vexing problem of having Polkadot’s and Kusama’s validators all hosted by AWS and similar providers
+with censorship capability over the network’s validators); as so, beyond the scope of the Milestone2, we think that progressing as fast
+as possible with the general roadmap of the Datagen network can be beneficial for the whole ecosystem.
+
+For all the reasons above, I would like to ask an integration of the amount due for the M2 deliverable of 9K USDC (from 12K to 21K
```

Comment:
Thanks @Lord-Nymphis I'm not sure what is meant here by "ask an integration of the amount due..." are you asking for us to pay for the milestone before it is completed? Or can you explain what you mean here?

Please note in our [terms and conditions](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/T%26Cs.md), that we aren't able to pay for milestones before the delivery is evaluated and accepted.

Reply:
@keeganquigley No, OfC I don't mean for the Milestone to be paid before is completed

Reply:
Thanks @Lord-Nymphis can you explain what you mean here?

Reply:
@keeganquigley I mean that, given that the end goal of the Datagen project (not the Milestone, which concerns only the RPC communication, between a Parachain and a Solochain, and not the Grant, which is just a PoC) is to provide cloud computing also for validators; the sooner we develop, the sooner we can have a positive eternality for other projects, including projects in the Polkadot ecosystem.
Obviously it doesn't mean that once we have developed this milestone the Datagen network is operational, but it means that we are one step faster.

Meaning that, by developing faster, we can achieve the goal of having a positive effect on other projects.

Reply:
@keeganquigley, I think `integration` should be `increase`. Then it makes sense.

Reply:
@semuelle Thanks, done. And corrected few other typos as well.